Library/Homebrew/extend/os/linux/install.rb

```diff
@@ -5,21 +5,32 @@ module Homebrew
   module Install
     module_function
 
-    DYNAMIC_LINKERS = [
-      "/lib64/ld-linux-x86-64.so.2",
-      "/lib64/ld64.so.2",
-      "/lib/ld-linux.so.3",
-      "/lib/ld-linux.so.2",
-      "/lib/ld-linux-aarch64.so.1",
-      "/lib/ld-linux-armhf.so.3",
-      "/system/bin/linker64",
-      "/system/bin/linker",
+    DYNAMIC_LINKERS = %w[
+      /lib64/ld-linux-x86-64.so.2
+      /lib64/ld64.so.2
+      /lib/ld-linux.so.3
+      /lib/ld-linux.so.2
+      /lib/ld-linux-aarch64.so.1
+      /lib/ld-linux-armhf.so.3
+      /system/bin/linker64
+      /system/bin/linker
     ].freeze
     private_constant :DYNAMIC_LINKERS
 
+    GCC_VERSION_SUFFIX = OS::LINUX_GCC_CI_VERSION.sub(/\.0$/, "").freeze
+
+    GCC_RUNTIME_LIBS = %w[
```

Comment:
@sjackman should this list also include `libatomic.so.1`?  It is a part of `gcc` and is included in the allowlist in https://github.com/Homebrew/brew/blob/61db6207d8d9cd652b182fc7f183ffebab4ac49c/Library/Homebrew/extend/os/linux/linkage_checker.rb.  Conversely, `libquadmath.so.0` is not in the allowlist in `linkage_checker.rb` but is in the list we are using here.  I think we should try to have consistency here across those two lists.

Reply:
Yes `GCC_RUNTIME_LIBS = %w[libatomic.so.1 libgcc_s.so.1 libgomp.so.1 libstdc++.so.6]` sounds good.
`SYSTEM_LIBRARY_ALLOWLIST` could potentially include `GCC_RUNTIME_LIBS`.
`libquadmath.so.0` is excluded from `SYSTEM_LIBRARY_ALLOWLIST`, because it's a Fortran library, and when a formula requires Fortran, the formula depends explicitly on `gcc` for macOS, and we do the same thing on Linux. We could consider changing the Linux behaviour, but I'd probably defer that until after the Homebrew 3.6.0 release.

Reply:
I think it does make sense for us not to link any Fortran libraries.  This seems like it might be a holdover from when we changed the default `gcc` on Linux to the most recent GCC rather than the CI version.  We don't need to change anything in `linkage_checker.rb`, just the libraries here.